plenum/test/replica/test_replica_removing.py

```diff
@@ -0,0 +1,157 @@
+                                          sdk_wallet_client,
+                                          chkFreqPatched):
+    """
+    1. Remove replica
+    2. Ordering
+    3. Check monitor and replicas count
+    """
+    node = txnPoolNodeSet[0]
+    start_replicas_count = node.replicas.num_replicas
+    index = start_replicas_count - 1
+    node.replicas.remove_replica(index)
+    sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle, sdk_wallet_client, 1)
+    looper.run(eventually(check_checkpoint_finalize, txnPoolNodeSet))
+    _check_replica_removed(node, start_replicas_count)
+    # trigger view change on all nodes for return backup replicas
+    for node in txnPoolNodeSet:
+        node.view_changer.on_master_degradation()
+    ensureElectionsDone(looper=looper, nodes=txnPoolNodeSet)
+
+
+def test_replica_removing_in_ordering(looper,
+                                      txnPoolNodeSet,
+                                      sdk_pool_handle,
+                                      sdk_wallet_client,
+                                      chkFreqPatched):
+    """
+    1. Start ordering (send pre-prepares on backup)
+    2. Remove replica
+    3. Finish ordering
+    4. Check monitor and replicas count
+    """
+    node = txnPoolNodeSet[0]
+    start_replicas_count = node.replicas.num_replicas
+    index = start_replicas_count - 1
+
+    slow_stasher = node.nodeIbStasher
+    with delay_rules(slow_stasher, cDelay()):
+        req = sdk_send_random_requests(looper,
+                                       sdk_pool_handle,
+                                       sdk_wallet_client,
+                                       1)
+        node.replicas.remove_replica(index)
+    sdk_get_replies(looper, req)
+    looper.run(eventually(check_checkpoint_finalize, txnPoolNodeSet))
+    _check_replica_removed(node, start_replicas_count)
+    # trigger view change on all nodes for return backup replicas
+    for node in txnPoolNodeSet:
+        node.view_changer.on_master_degradation()
+    ensureElectionsDone(looper=looper, nodes=txnPoolNodeSet)
+
+
+def test_replica_removing_after_ordering(looper, txnPoolNodeSet, sdk_pool_handle, sdk_wallet_client):
+    """
+    1. Ordering
+    2. Remove replica
+    3. Check monitor and replicas count
+    """
+    node = txnPoolNodeSet[0]
+    start_replicas_count = node.replicas.num_replicas
+    sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle, sdk_wallet_client, 1)
+    looper.run(eventually(check_checkpoint_finalize, txnPoolNodeSet))
+    index = start_replicas_count - 1
+    node.replicas.remove_replica(index)
+    _check_replica_removed(node, start_replicas_count)
+    # trigger view change on all nodes for return backup replicas
+    for node in txnPoolNodeSet:
+        node.view_changer.on_master_degradation()
+    ensureElectionsDone(looper=looper, nodes=txnPoolNodeSet)
+
+
+def _check_replica_removed(node, start_replicas_count):
```

Comment:
This method should take an instance ID. Also it should check that items related to the replica are removed from all the relevant collections and it should not verify conditions not related to removing the replica (such as master degradation and requests freeing).

Reply:
Added check that replica removed from all relevant collections and changed parameters list. But I think, we should check things as a master degraded in _check_replica_removed() because it's a proof that replica was successfully removed without negative effects.

Reply:
We should check this but not in `_check_replica_removed` method since it is not its responsibility according to its name.